haskell_scotty/ClientRepo.hs

```diff
@@ -16,3 +18,19 @@ getClients = do
 
 instance QueryResults Client where
   convertResults fs vs = Client $... zip fs vs
+
+saveClient :: Client -> IO Client
+saveClient client = do
+  conn <- connectDB
+  execute conn "INSERT INTO clients \
+    \(name,address,city,state,zip,country,email,contact,phone,created_at,updated_at) values \
+    \(?,?,?,?,?,?,?,?,?,?,?)" client
+  results <- query_ conn "SELECT LAST_INSERT_ID()"
+  let [Only clientId] = results
+  return client { Client.id = clientId }
+
+instance QueryParams Client where
+  renderParams
+    (Client _ name address city state zip country email contact phone created_at updated_at) =
+      [render name, render address, render city, render state, render zip, render country, render email, render contact, render phone, render created_at, render updated_at]
```

Comment:
No, there's really no lists of elements of heterogenous types. The `Param` typeclass is your  we-don't-care-as-long-as-it-can-be-rendered type thingy, but it's not really a type, so you can't make lists of them unless all your elements are really of the same type: even then, you'll have trouble using the list because the type system loses track of what's going on. Normally you would make a union type and make lists of elements.